Improve docs for custom events (#1587)
pytorch/ignite: docs/source/concepts.rst

        CUSTOM_STARTED = 'custom_started'
        CUSTOM_COMPLETED = 'custom_completed'

    engine.register_events(
        *CustomEvents,

>         event_to_attr={

Reviewer: This part is not required actually. I'd put everything in the following way:
- simple custom events without filtering (current doc)
- custom events with filtering:
   - engine.state should have corresponding attributes
   - event_to_attr is required when `register_events`
   - should increase counters (subject to be removed)

    @engine.on(Events.STARTED)
    def fire_custom_events(engine):
         engine.fire_event(CustomEvents.CUSTOM_STARTED)

>          engine.state.custom_started += 1  # increase state counter for this event (required)

Reviewer: Let's say this is required for filtering only and can be omitted if no filtering is done. And another note to add is that this is a temporary solution and subject to change in future releases.

   See the source code of :class:`~ignite.contrib.engines.create_supervised_tbptt_trainer` for an example of usage of
   custom events.

> If you want to use filtering with custom events (e.g. with `CallableEventWithFilter`), you need to do 3 more things:

Reviewer: ```suggestion
If you want to use filtering with custom events (e.g. with ``CustomEvents.CUSTOM_STARTED(every=5)``), you need to do 3 more things:
```

> If you want to use filtering with custom events (e.g. with `CallableEventWithFilter`), you need to do 3 more things:

Reviewer: Can you also please add a subsection to the line 229 : 

    Custom events
    ``````````````

Author: Done!

> - `engine.state` should have corresponding attributes for the events, e.g. `engine.state.custom_started`

Reviewer: ```suggestion
- ``engine.state`` should have corresponding attributes for the events, e.g. ``engine.state.custom_started``
```

> - you need to pass a dict `event_to_attr` to `register_events`, which maps between events and state attributes, e.g.

Reviewer: ```suggestion
- you need to pass a dict `event_to_attr` to :meth:`~ignite.engine.engine.Engine.register_events`, which maps between events and state attributes, e.g.
```

.. code-block:: python

>     event_to_attr={
>             CustomEvents.CUSTOM_STARTED: "custom_started",
>             CustomEvents.CUSTOM_COMPLETED: "custom_completed",
>         }

Reviewer: ```suggestion
    event_to_attr = {
        CustomEvents.CUSTOM_STARTED: "custom_started",
        CustomEvents.CUSTOM_COMPLETED: "custom_completed",
    }
```

> - you should increase the counter for the event whenever you fire the event, e.g. `engine.state.custom_started += 1`

Reviewer: ```suggestion
- you should increase the counter for the event whenever you fire the event, e.g. ``engine.state.custom_started += 1``
```

> .. Note ::

Reviewer: ```suggestion
.. warning::
```

   See the source code of :class:`~ignite.contrib.engines.create_supervised_tbptt_trainer` for an example of usage of
   custom events.

> If you want to use filtering with custom events (e.g. with ``CustomEvents.CUSTOM_STARTED(every=5)``), you need to do 3 more things:

Reviewer: ```suggestion
If you want to use filtering with custom events (e.g. ``CustomEvents.CUSTOM_STARTED(every=5)``), you need to do 3 more things:
```
